### ISSUE TITLE
H2O randomForest and GBM models DO support missing values

### ISSUE THREAD

**Issue by kdpsingh** [mlr-org/mlr#1690]
Hello,

I wanted to recommend adding the "missings" property to H2O's randomForest and GBM models as both of these models internally deal with missing values without imputation. The other H2O models will skip over missing values and correctly do not have the "missings" property. However, as you can see from the documentation below, H2O RF and GBM do support NAs. I can confirm from changing the missings property for H2O's RF on my machine that everything works properly and the response does not contain NAs.

Thanks!

## H2O documentation page for random forest: http://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/drf.html

**How does the algorithm handle missing values during training?**
Missing values are interpreted as containing information (i.e., missing for a reason), rather than missing at random. During tree building, split decisions for every node are found by minimizing the loss function and treating missing values as a separate category that can go either left or right.

**How does the algorithm handle missing values during testing?**
During scoring, missing values follow the optimal path that was determined for them during training (minimized loss function).

## H2O documentation for GBM: http://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/gbm-faq/missing_values.html

**How does the algorithm handle missing values during training?**
Missing values are interpreted as containing information (i.e., missing for a reason), rather than missing at random. During tree building, split decisions for every node are found by minimizing the loss function and treating missing values as a separate category that can go either left or right.

So how does the algorithm determine what goes left or right? The algo decides which feature and level/number value to split on its decision to make things go left or right is purely based on value (i.e. everything less than the split point value goes left, and everything greater and equal goes right). This means that missing values can go left or right in the same way as any other categorical because NAs are now seen as a new categorical.

A note on nbins_cats: This parameter specifies the number of bins to use for non-Na categoricals, it does not bin missing values. Regardless of the number of bins a user chooses for nbins_cats, an additional bin will be created specifically for the missing values.

**How does the algorithm handle missing values during testing?**
During scoring, missing values follow the optimal path that was determined for them during training (minimized loss function).

**Comment by mllg** [comment]
Thanks for the report!

**Comment by kdpsingh** [comment]
Thanks for merging. However, as per my original note, only h2o GBM and h2o randomForest support missings. The other h2o models (including GLM) currently don't support missing values -- there is an interface to perform mean imputation via h2o but I wouldn't count that as supporting missing values.

I would suggest removing the missing values property for the h2o GLM and leaving it in for GBM and RF only. Thanks!

**Comment by berndbischl** [comment]
@kdpsingh i have fixed this in master now. thank you for having a look twice and being so precise.

@mllg lets watch out for this, we really dont want to become inconsistent here.
actually: why did this pass on travis? learners with missing are tested with missing values?
if h20 glm does not support this, there should have been an error?

@larskotthoff calling the unit test general to this thread.

**Comment by berndbischl** [comment]
please respond in #1724
